numpy/core/tests/test_deprecations.py

```diff
@@ -618,3 +618,25 @@ def test_not_deprecated(self):
             self.assert_not_deprecated(round, args=(scalar,))
             self.assert_not_deprecated(round, args=(scalar, 0))
             self.assert_not_deprecated(round, args=(scalar,), kwargs={'ndigits': 0})
+
+
+class TestIncorrectAdvancedIndexWithEmptyResult(_DeprecationTestCase):
+    # 2020-04-02, NumPy 1.19.0
+    message = "Out of bound index found. This was previously ignored"
+
+    @pytest.mark.parametrize("index", [([3, 0],), ([0, 0], [3, 0])])
+    def test_empty_subspace(self, index):
+        # Test for both a single and two/multiple advanced indices. These
+        # This will raise an IndexError in the future.
+        arr = np.ones((2, 2, 0))
+        self.assert_deprecated(arr.__getitem__, args=(index,))
+        self.assert_deprecated(arr.__setitem__, args=(index, []))
+
+    def test_empty_index_broadcast(self):
+        arr = np.ones((2, 2, 2))
+
+        index = ([[3], [2]], [])  # broadcast to an empty result.
+        self.assert_deprecated(arr.__getitem__, args=(index,))
+        self.assert_deprecated(arr.__setitem__,
+                               args=(index, np.empty((2, 0, 2))))
```

Comment:
Apologies for the noise on the release note - I now see you have a test for it after all.

In my opinion, this should not error, in the same way that `np.array([]) / 0` should not emit a floating-point warning

Reply:
Yes, I think I concur now, I did not think carefully the differences in the two cases.